libs/positions/src/lib/positions-table.tsx

```diff
@@ -113,234 +124,299 @@ export const AmountCell = ({ valueFormatted }: AmountCellProps) => {
-        headerName={t('Mark price')}
-        field="markPrice"
-        type="rightAligned"
-        cellRenderer="PriceFlashCell"
-        valueFormatter={({
-          value,
-          data,
-        }: PositionsTableValueFormatterParams & {
-          value: Position['markPrice'];
-        }) => {
-          if (!data) {
-            return undefined;
-          }
-          if (
-            data.marketTradingMode ===
-            MarketTradingMode.TRADING_MODE_OPENING_AUCTION
-          ) {
-            return '-';
-          }
-          return addDecimalsFormatNumber(
-            value.toString(),
-            data.marketDecimalPlaces
-          );
-        }}
-      />
-      <AgGridColumn
-        headerName={t('Entry price')}
-        field="averageEntryPrice"
-        headerComponentParams={{
-          template:
-            '<div class="ag-cell-label-container" role="presentation">' +
-            `  <span>${t('Liquidation price (est)')}</span>` +
-            '  <span ref="eText" class="ag-header-cell-text"></span>' +
-            '</div>',
+const ButtonCell = ({
+  onClick,
+  data,
+}: {
+  onClick: (position: Position) => void;
+  data: Position;
+}) => {
+  return <Button onClick={() => onClick(data)}>{t('Close')}</Button>;
+};
+
+const EmptyCell = () => '';
+
+export const PositionsTable = forwardRef<AgGridReact, Props>(
+  ({ onClose, ...props }, ref) => {
+    return (
+      <AgGrid
+        style={{ width: '100%', height: '100%' }}
+        overlayNoRowsTemplate="No positions"
+        getRowId={getRowId}
+        rowHeight={34}
+        ref={ref}
+        defaultColDef={{
+          flex: 1,
+          resizable: true,
         }}
-        flex={2}
-        cellRenderer="ProgressBarCell"
-        valueFormatter={({
-          data,
-        }: PositionsTableValueFormatterParams):
-          | PriceCellProps['valueFormatted']
-          | undefined => {
-          if (!data) {
-            return undefined;
-          }
-          const min = BigInt(data.averageEntryPrice);
-          const max = BigInt(data.liquidationPrice);
-          const mid = BigInt(data.markPrice);
-          const range = max - min;
-          return {
-            low: addDecimalsFormatNumber(
-              min.toString(),
-              data.marketDecimalPlaces
-            ),
-            high: addDecimalsFormatNumber(
-              max.toString(),
+        components={{ PriceFlashCell, ProgressBarCell }}
+        {...props}
+      >
+        <AgGridColumn
+          headerName={t('Market')}
+          field="marketName"
+          cellRenderer={MarketNameCell}
+          valueFormatter={({
+            value,
+          }: PositionsTableValueFormatterParams & {
+            value: Position['marketName'];
+          }) => {
+            if (!value) {
+              return undefined;
+            }
+            // split market name into two parts, 'Part1 (Part2)'
+            const matches = value.match(/^(.*)\((.*)\)\s*$/);
```

Comment:
chore (blocking): I'm not sure it'll be safe to assume that the market name will be in this format _and_ that the value in the parentheses will always be the expiry date. 

@barnabee any guidance here?

I think we should be just rendering the name as is for now. And revisit once we have a reliable way to retrieve expiry datetimes (from metadata).

Finally if we don't get a match so the second row is omitted, it might make the table ugly as the spacing between the rows will be inconsistent

Reply:
So far market names looks like this
```
Apple Monthly (30 Jun 2022)
AAVEDAI Monthly (30 Jun 2022)
BTCUSD Monthly (30 Jun 2022)
AAVEDAI Monthly (30 Jun 2022)
Tesla Quarterly (30 Jun 2022)
Tesla Quarterly (30 Jun 2022)
OP/USD AUG-SEP22 - Incentive.
UNIDAI Monthly (30 Jun 2022)
AAVEDAI Monthly (30 Jun 2022)
Apple Monthly (30 Jun 2022)
ETHBTC Quarterly (30 Jun 2022)
UNIDAI Monthly (30 Jun 2022)
Tesla Quarterly (30 Jun 2022)
BTCUSD Monthly (30 Jun 2022)
ETHBTC Quarterly (30 Jun 2022)
Apple Monthly (30 Jun 2022)
BTCUSD Monthly (30 Jun 2022)
ETHBTC Quarterly (30 Jun 2022)
UNIDAI Monthly (30 Jun 2022)
CELUSD (June 2022)
UNIUSD (Incentive)
```

If scheme will change we can update regexp. I will modify it to support "ABC (XYZ)" and "ABC - XYZ"